### PR TITLE
Update CodeCov report configuration

### LINKED ISSUE #3176: [CI] [Failing Test] Upload coverage to CodeCov (ui) on UI Integration Tests
#### Current Behavior

The step of **Upload coverage to CodeCov (UI)** on the Integration Tests job is falling. The error message is:

```
==> Uploading reports
    url: https://codecov.io
    query: branch=&commit=58665caa48dadebecbb24fb4c12398099ca2b2bf&build=&build_url=&name=&tag=&slug=layer5io%2Fmeshery&service=&flags=integrationtests&pr=&job=&cmd_args=n,F,Z
->  Pinging Codecov
https://codecov.io/upload/v4?package=bash-1.0.3&token=secret&branch=&commit=58665caa48dadebecbb24fb4c12398099ca2b2bf&build=&build_url=&name=&tag=&slug=layer5io%2Fmeshery&service=&flags=integrationtests&pr=&job=&cmd_args=n,F,Z

Could not determine repo and owner
400
Error: Codecov failed with the following error: The process 'bash' failed with exit code 1
```
_See: https://github.com/layer5io/meshery/pull/3170/checks?check_run_id=2781382844#step:6:53_

#### Desired Behavior

All the tests should always be green on master.

#### Reference:
- [Meshery Build & Release Strategy](https://docs.google.com/document/d/11nAxYtz2SUusCYZ0JeNRrOLIxkgmmbUVWz63MBZV2oE/edit#)

### PR DESCRIPTION
**Pull request by ramrodo**
**Description**

This PR:
- Specifies flags for the different tests in the project (unit, integration, and e2e tests)
- Specifies the folders to be measured for each flag. More info in [flags docs](https://docs.codecov.io/docs/flags)
- Adds `carryforward` option to carry the previous coverage for the other tests. More info in [carryforward docs](https://docs.codecov.io/docs/carryforward-flags)

This PR fixes #3176

**Notes for Reviewers**

<s>NOT MERGE until this PR is merged: https://github.com/layer5io/meshery/pull/3179</s>

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

**Comment by ramrodo** [comment]
Blocked until this PR is merged: https://github.com/layer5io/meshery/pull/3179

**Comment by leecalcote** [comment]
PR #3179 is merged. ✅

**Review by leecalcote — APPROVED** [review]
LGTM

**Pull request merged**
